network-api/networkapi/wagtailpages/pagemodels/mixin/foundation_metadata.py

```diff
@@ -33,23 +37,40 @@ def get_meta_description(self):
 
         return self.default_description
 
+    def set_default_social_share_tag_and_image(self):
+        # necessary because we're going to reassign them, rather than
+        # assign same-named vars in local scope:
+        global default_social_share_tag, default_social_share_image
+
+        # get the tag with name "social share image"
+        default_share_tag_name = 'social share image'
+        tag, create = Tag.objects.get_or_create(name=default_share_tag_name)
+        default_social_share_tag = tag
+
+        # then find an image in the CMS that uses that tag:
+        default_social_share_image = None
+        default_social_share_images = Image.objects.filter(tags=default_social_share_tag)
+        if default_social_share_images.count() > 0:
+            default_social_share_image = default_social_share_images.first()
```

Comment:
```suggestion
        find_default_social_share_image = Image.objects.filter(tags=default_social_share_tag).first()
        if find_default_social_share_image:
            default_social_share_image = find_default_social_share_image
```
☝️ This way the .filter() will stop searching through the DB when it finds the first instance

Reply:
I rewrote it _from_ that because it was in a try/except. What happens when you call `first()` on an empty queryset?